ergo-mempool-sync/src/client/node.rs

```diff
@@ -1,9 +1,55 @@
 use ergo_chain_sync::client::model::FullBlock;
 use ergo_lib::chain::transaction::Transaction;
+use async_trait::async_trait;
+use ergo_lib::ergo_chain_types::{BlockId, Header};
+use isahc::{AsyncReadResponseExt, HttpClient};
+use ergo_chain_sync::client::types::{Url, with_path};
 
-#[async_trait::async_trait(?Send)]
+use crate::client::models::ApiInfo;
+
+#[async_trait::async_trait(? Send)]
 pub trait ErgoNetwork {
     async fn fetch_mempool(&self, offset: usize, limit: usize) -> Vec<Transaction>;
```

Comment:
I suggest to change the return type to `Result<Vec<Transaction>, ClientError>`, so we can catch any errors when talking to the node.

Reply:
By the way, I'm making a similar change to `ErgoNetwork::get_block_at` in #25 (see https://github.com/spectrum-finance/spectrum-offchain-ergo/blob/91ee0961e03a51d4c56c12215206aaa31548dd2e/ergo-chain-sync/src/client/node.rs#L44)